### PR TITLE
Add test cases for search by add date usecase

### LINKED ISSUE #48: 8 - Test Search Foods by Addition Date
# Original Task Description

Search Foods by Add Date: Use Kotlin’s Date class to represent the date in the meal entity. Let the user input a date and return a list of IDs and names of meals added on that date. The user should be able to view details of a specific meal by entering its ID. Handle exceptions for:
- Incorrect date format.
- No meals were found for the given date. Ensure different exceptions are used for both cases.

# Task overview

Test and refactor the corresponding usecase.

Please don't introduce any changes in the UI or the dependency injection modules, leave them for last to avoid conflicts.

### PR DESCRIPTION
**Pull request by yusufnasserdev**
Closes #48

### CODE REVIEW

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt [hunk 1]
```diff
@@ -12,14 +12,14 @@ class GetMealsByDateUseCase(private val repository: MealRepository) {
 
         val targetDate: LocalDate = try {
             LocalDate.parse(inputDate, formatter)
-        } catch (e: Exception) {
+        } catch (_: Exception) {
             throw IllegalArgumentException("⚠️ Incorrect date format. Use yyyy-MM-dd.")
```
Comment (AmeerAmjed) [inline src/main/kotlin/useCase/GetMealsByDateUseCase.kt]: Can be improved by creating a custom exception and setting the message in the UI.

File: src/test/kotlin/useCase/GetMealsByDateUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        )
+    }
+
+    private fun listOfMealsWithIncorrectlyFormattedDates(): List<Meal> {
+        return listOf(
+            defaultMeal(submitted = "2024-12-31"),
+            defaultMeal(submitted = "12-11-2023"),
+            defaultMeal(submitted = "2024-12-31")
+        )
+    }
```
Comment (AmeerAmjed) [inline src/test/kotlin/useCase/GetMealsByDateUseCaseTest.kt]: We can move these private functions to the end of the class.

File: src/test/kotlin/useCase/GetMealsByDateUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    @BeforeEach
+    fun setup() {
+        mealRepository = mockk(relaxed = true)
+        getMealsByDateUseCase = GetMealsByDateUseCase(mealRepository)
+    }
+
```
Comment (AmeerAmjed) [inline src/test/kotlin/useCase/GetMealsByDateUseCaseTest.kt:28]: You can add a test case for when inputDate is empty.

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt (hunk 1 above)
Comment (yusufnasserdev) [reply src/main/kotlin/useCase/GetMealsByDateUseCase.kt]: The message is kept as is for logs, it's the UI role to use this message or a different one depending on the user's region or language (if applies).

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt [hunk 4]
```diff
@@ -12,20 +14,20 @@ class GetMealsByDateUseCase(private val repository: MealRepository) {
 
         val targetDate: LocalDate = try {
             LocalDate.parse(inputDate, formatter)
-        } catch (e: Exception) {
-            throw IllegalArgumentException("⚠️ Incorrect date format. Use yyyy-MM-dd.")
+        } catch (_: Exception) {
+            throw IllegalDateFormatException("Incorrect date format. Use yyyy-MM-dd.")
```
Comment (AmeerAmjed) [inline src/main/kotlin/useCase/GetMealsByDateUseCase.kt:18]: When using a custom exception, there's no need to pass a message to the UI. The UI should handle the exception and convert it into a message.

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt [hunk 5]
```diff
@@ -12,20 +14,20 @@ class GetMealsByDateUseCase(private val repository: MealRepository) {
...
                 LocalDate.parse(meal.submitted, formatter) == targetDate
-            } catch (e: Exception) {
+            } catch (_: Exception) {
                 false
             }
         }
 
         if (matchedMeals.isEmpty()) {
-            throw NoSuchElementException("❌ No meals found on $inputDate.")
+            throw NoSuchMealException("No meals found on $inputDate.")
```
Comment (AmeerAmjed) [inline src/main/kotlin/useCase/GetMealsByDateUseCase.kt:30]: When using a custom exception, there's no need to pass a message to the UI. The UI should handle the exception and convert it into a message.

**Review by Al-Taie — APPROVED** [review]

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt
Comment (AmeerAmjed) [inline src/main/kotlin/useCase/GetMealsByDateUseCase.kt:1]: The UI should catch these exceptions.

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt (hunk 4 above)
Comment (yusufnasserdev) [reply src/main/kotlin/useCase/GetMealsByDateUseCase.kt:18]: The message is for the application logs, the UI is free to ignore it and use a different one.

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt (hunk 5 above)
Comment (yusufnasserdev) [reply src/main/kotlin/useCase/GetMealsByDateUseCase.kt:30]: The message is for the application logs, the UI is free to ignore it and use a different one.

File: src/main/kotlin/useCase/GetMealsByDateUseCase.kt
Comment (yusufnasserdev) [reply src/main/kotlin/useCase/GetMealsByDateUseCase.kt:1]: I'm a little confused, this is quite literally contradicts the message above, so which should I follow?

**Review by AmeerAmjed — APPROVED** [review]
